Update for new class based makers

PascalPixel opened the pull request.

PascalPixel committed:
Update index.ts

PascalPixel committed:
Update index.ts

Burzo commented on index.ts:
```diff
@@ -33,18 +34,19 @@ class ElectronForgeSslCodeSignPlugin extends PluginBase<ConfigTypes> {
-				"@electron-forge/maker-squirrel",
-		) as IForgeResolvableMaker
+		const _squirrelMaker = forgeConfig.makers.find(
+			(maker) => "name" in maker && !maker.name.includes("squirrel"),
```
There is probably a typo here, I assume `!maker.name.includes("squirrel")` should be `maker.name.includes("squirrel")`.

Burzo reviewed (changes requested):
How do you pass the config in your case - could you give me an example please? Or link me the docs? In 7.x Forge you always do it like this:

```
{
	name: "@electron-forge/maker-squirrel",
	config: {
		setupIcon: path.join("build-assets", flavor, "logo.ico"),
		loadingGif: path.join("build-assets", flavor, "installing.gif"),
		iconUrl: process.env.WINDOWS_APP_ICON_URL,
		setupExe: `${safeAppNameWithVersion}-windows.exe`,
		name: safeAppName,
	},
},
```

PascalPixel committed:
Update index.ts

PascalPixel committed:
Update index.ts

PascalPixel committed:
Update index.ts

PascalPixel committed:
Update index.ts

PascalPixel committed:
Update index.ts

PascalPixel commented:
> How do you pass the config in your case - could you give me an example please? Or link me the docs? In 7.x Forge you always do it like this:
> 
> ```
> {
> 	name: "@electron-forge/maker-squirrel",
> 	config: {
> 		setupIcon: path.join("build-assets", flavor, "logo.ico"),
> 		loadingGif: path.join("build-assets", flavor, "installing.gif"),
> 		iconUrl: process.env.WINDOWS_APP_ICON_URL,
> 		setupExe: `${safeAppNameWithVersion}-windows.exe`,
> 		name: safeAppName,
> 	},
> },
> ```

aye it seems like the config isn't coming through (haven't been resolved yet, which is odd for a postMake hook but ok...)

- I have fixed the types
- I have switched to using the artifacts to find the .exe the same as you find the .nupkg

Obviously this would break if there are other .exe's in the artifacts but I can't figure out how else to get the setupExe path at the moment...

Burzo commented:
> > How do you pass the config in your case - could you give me an example please? Or link me the docs? In 7.x Forge you always do it like this:
> > ```
> > {
> > 	name: "@electron-forge/maker-squirrel",
> > 	config: {
> > 		setupIcon: path.join("build-assets", flavor, "logo.ico"),
> > 		loadingGif: path.join("build-assets", flavor, "installing.gif"),
> > 		iconUrl: process.env.WINDOWS_APP_ICON_URL,
> > 		setupExe: `${safeAppNameWithVersion}-windows.exe`,
> > 		name: safeAppName,
> > 	},
> > },
> > ```
> 
> aye it seems like the config isn't coming through (haven't been resolved yet, which is odd for a postMake hook but ok...)
> 
> * I have fixed the types
> * I have switched to using the artifacts to find the .exe the same as you find the .nupkg
> 
> Obviously this would break if there are other .exe's in the artifacts but I can't figure out how else to get the setupExe path at the moment...

I'm still having trouble understanding how you set up your project, could you please let me know how your forge config (`forge.config.js`) file looks like?

It's hard to properly review without this sort of info.

PascalPixel commented:
Here's what the typescript templates produce for the forge config;

```forge.config.ts
import type { ForgeConfig } from '@electron-forge/shared-types';
import { MakerSquirrel } from '@electron-forge/maker-squirrel';
import { MakerZIP } from '@electron-forge/maker-zip';
import { MakerDeb } from '@electron-forge/maker-deb';
import { MakerRpm } from '@electron-forge/maker-rpm';
import { VitePlugin } from '@electron-forge/plugin-vite';
import { FusesPlugin } from '@electron-forge/plugin-fuses';
import { FuseV1Options, FuseVersion } from '@electron/fuses';

const config: ForgeConfig = {
  packagerConfig: {
    asar: true,
  },
  rebuildConfig: {},
  makers: [new MakerSquirrel({}), new MakerZIP({}, ['darwin']), new MakerRpm({}), new MakerDeb({})],
  plugins: [
    new VitePlugin({
      // `build` can specify multiple entry builds, which can be Main process, Preload scripts, Worker process, etc.
      // If you are familiar with Vite configuration, it will look really familiar.
      build: [
        {
          // `entry` is just an alias for `build.lib.entry` in the corresponding file of `config`.
          entry: 'src/main.ts',
          config: 'vite.main.config.ts',
        },
        {
          entry: 'src/preload.ts',
          config: 'vite.preload.config.ts',
        },
      ],
      renderer: [
        {
          name: 'main_window',
          config: 'vite.renderer.config.ts',
        },
      ],
    }),
    // Fuses are used to enable/disable various Electron functionality
    // at package time, before code signing the application
    new FusesPlugin({
      version: FuseVersion.V1,
      [FuseV1Options.RunAsNode]: false,
      [FuseV1Options.EnableCookieEncryption]: true,
      [FuseV1Options.EnableNodeOptionsEnvironmentVariable]: false,
      [FuseV1Options.EnableNodeCliInspectArguments]: false,
      [FuseV1Options.EnableEmbeddedAsarIntegrityValidation]: true,
      [FuseV1Options.OnlyLoadAppFromAsar]: true,
    }),
  ],
};

export default config;
```

It uses a class-based maker config in order to properly provide types, without this PR `electron-forge-code-sign-plugin` only works with the object-based maker configs as shown in your comment, so to make it work with the class-based configs, I made this PR, as they resolve a little differently

PascalPixel commented on index.ts:
```diff
@@ -33,15 +34,17 @@ class ElectronForgeSslCodeSignPlugin extends PluginBase<ConfigTypes> {
+			forgeConfig.makers.find(
+				(maker) =>
+					"name" in maker && maker.name === "@electron-forge/maker-squirrel",
+			)
```
squirrelResolvableMaker here resolves the object-based maker
squirrelMaker resolves the class-based maker
